lib/content-services/src/lib/tree/components/tree.component.ts

```diff
@@ -0,0 +1,250 @@
+        });
+    }
+
+    /**
+     * Checks if node is LoadMoreNode node
+     *
+     * @param node node to be checked
+     * @returns boolean
+     */
+    public isLoadMoreNode(_: number, node: T): boolean {
+        return node.nodeType === TreeNodeType.LoadMoreNode;
+    }
+
+    /**
+     * Checks if tree is empty
+     *
+     * @returns boolean
+     */
+    public isEmpty(): boolean {
+        return this.treeService.isEmpty();
+    }
+
+    /**
+     * Returns action icon based on expanded/collapsed node state.
+     *
+     * @param node node to be checked
+     * @returns collapse or expand icon
+     */
+    public expandCollapseIconValue(node: T): string {
+        return this.treeService.treeControl.isExpanded(node) ? this.collapseIcon : this.expandIcon;
+    }
+
+    /**
+     * Refreshes the tree, root nodes are reloaded, tree selection is cleared.
+     *
+     * @param skipCount Number of root nodes to skip.
+     * @param maxItems Maximum number of nodes returned from Observable.
+     */
+    public refreshTree(skipCount?: number, maxItems?: number): void {
+        this.loadingRootSource.next(true);
+        this.treeNodesSelection.clear();
+        this.treeService.getSubNodes('-root-', skipCount, maxItems).subscribe((response: TreeResponse<T>) => {
+            this.treeService.treeNodes = response.entries;
+            this.treeNodesSelection.deselect(...response.entries);
+            this.paginationChanged.emit(response.pagination);
+            this.loadingRootSource.next(false);
+        });
+    }
+
+    /**
+     * Collapses or expanding the node based on its current state
+     *
+     * @param node node to be collapsed/expanded
+     */
+    public expandCollapseNode(node: T): void {
+        if (this.treeService.treeControl.isExpanded(node)) {
+            this.treeService.collapseNode(node);
+        } else {
+            node.isLoading = true;
+            this.treeService.getSubNodes(node.id, 0, this.userPreferenceService.paginationSize).subscribe((response: TreeResponse<T>) => {
+                this.treeService.expandNode(node, response.entries);
+                this.paginationChanged.emit(response.pagination);
+                node.isLoading = false;
+                if (this.treeNodesSelection.isSelected(node)) {
+                    //timeout used to update nodeCheckboxes query list after new nodes are added so they can be selected
+                    setTimeout(() => {
+                        this.treeNodesSelection.select(...response.entries);
+                    });
+                }
+            });
+        }
+    }
+
+    /**
+     * Loads more subnode for a given parent node
+     *
+     * @param node parent node
+     */
+    public loadMoreSubnodes(node: T): void {
+        node.isLoading = true;
+        const parentNode: T = this.treeService.getParentNode(node.parentId);
+        this.treeService.treeNodes.splice(this.treeService.treeNodes.indexOf(node), 1);
```

Comment:
does not look right, you mutate the service variables from within the component, maybe worth having a public method in the service that does that

Reply:
Added public method for this